### DISCUSSION TITLE
Best practice for hosting Docker instance of Semaphore

### DISCUSSION THREAD

**Discussion by nebularnoise** [semaphoreui/semaphore#2265] [Q&A]
Hi !

I have a virtualization server with a bunch of services on it, some running in VMs, most running on Docker within a VM.
So far, I've been running ansible playbooks from my workstation, but I'm interested in using Semaphore.

My first "naive" intuition would be to run Semaphore in a docker container running on the aforementioned VM, like everything else on my LAN. However, that would mean I wouldn't be able to manage the docker host in any way, nor the hypervisor, unless I'm missing something.

What would you recommend ? Should I try and run Semaphore on an entirely different machine ?
Thanks

**Comment by sonor3000** [comment]
Semaphore is connecting via ssh to the target host in an inventory. As
long as this hosts can be reached via ssh all is fine.

We are also using Semaphore as a docker stack and it is no problem to
connect to VM where all our d docker deployments are running, the
hypervisor hosts os our Proxmox cluster  and also to other real hardware
hosts.

as long as you can reach your target hosts via ssh all is fine IMHO.

**Comment by sonor3000** [comment]
OK, there is one exception...

Ofcourse the host where the Semaphore docker stack is running should not be rebootet or something like that during Semaphore is working. Also restarts of docker have to be avoided during Semaphore is executing tasks.

**Reply by nebularnoise** [reply]
That exception is precisely why I'm asking how other people do it, I'd like to be able to restart systems if needed.

If it comes to it, I can live with that caveat, but if there's a better solution, I'm all ears.

**Reply by MastrUsr** [reply]
Did you find a good solution to this? I have the same question..

**Reply by nebularnoise** [reply]
Unfortunately not, though I have not explored the issue further.
I guess the simplest solution would be to use a separate docker host.

**Reply by MastrUsr** [reply]
I decided to run a separate vm to update all the others. Haven't decided if I'll keep updating semaphore manually or run another instance to update it..
